cpp/src/arrow/compute/exec/options.h

```diff
@@ -523,7 +523,8 @@ class ARROW_EXPORT AsofJoinNodeOptions : public ExecNodeOptions {
   ///
   /// \see `Keys` for details.
   std::vector<Keys> input_keys;
-  /// \brief Tolerance for inexact "on" key matching.  Must be non-negative.
+  /// \brief Tolerance for inexact "on" key matching. Positive (resp. negative) tolerance
+  /// is interpreted as a future (resp. past) as-of-join.
```

Comment:
The match condition is actually (now) reversed. I committed a more detailed doc.

Reply:
Out of curiosity, why is it reversed?  Doesn't this put us at odds with pandas which does backwards by default?

Reply:
I wouldn't say the PR is at odds because it takes the tolerance as a single parameter, a positive or negative number, whereas pandas takes the tolerance as two parameters, a positive number and a direction, even though the direction defaults to backwards. We could make this PR take the tolerance as two parameters if we want strict consistency. @icexelloss requested this standardization and should be able to explain the context.